src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -89,7 +89,8 @@ class RelayDefaultNetworkLayer {
         }
       }).catch(
         error => request.reject(error)
-      )
+      ),
+      request
```

Comment:
The problem isn't context. There're two promises: 1 - request ( argument ), 2 -this._sendQuery(request). And the function returnes the second promise, but works with the first.